### ISSUE TITLE
QITE Inconsistencies

### ISSUE THREAD

**Issue by JonathonMisiewicz** [evangelistalab/qforte#67]
After more poking around, I'm not convinced that QITE follows the same algorithm as [Mario and Garnet's paper](https://www.nature.com/articles/s41567-019-0704-4). QForte's [algorithm to construct b](https://github.com/evangelistalab/qforte/blob/master/src/qforte/ite/qite.py#L235) computes it using eq. 5 of that paper, with the denominator approximate per eq. 3. As best as I can tell, that isn't what Mario and Garnet did. The b vector in [https://github.com/mariomotta/QITE/blob/master/code_v4/qite.py#L71-L86](their code) is computed using eq. 9 of their SI. The approach of using eq. 3 and 5 of the main paper is a first order approximation to that, and in general, the two are different.

Is there a reason why we implement the eq. 3/5 strategy? I believe that eq. 9-SI is the correct one here.

There may be _other_ issues. It'll take some time to go through them.

**Comment by JonathonMisiewicz** [comment]
I'm going to be taking this issue as I read over some of Minnich's papers.

**Comment by JonathonMisiewicz** [comment]
The tale is more complicated than I first thought. They used _different approximations_ in their quantum simulator compared to their noiseless simulator. The algorithm they used on their noiseless system involves non-Trotterized exponentials.

I'm still not sure the QForte implementation matches the original implementation. The [basic structure of a single QITE step](https://github.com/mariomotta/QITE/blob/master/QPU_implementation/qite.py#L39-L43) is to loop over all terms in the Hamiltonian, compute the step for each, and update the wavefunction gate for each. In the QForte loop, we have a "step-merged" version, where we use the entire Hamiltonian, Trotterizing when necessary.

@nstair, does this sound right to you? If we're agreed about the theory, I can implement the non-step-merged version, keeping the step-merged version.

**Comment by nstair** [comment]
@JonathonMisiewicz  you are correct. The current implementation in QForte is closer to the step-merged version. You are welcome to implement the non-step-merged version, but it becomes challenging to rationalize 'inexact QITE' for non k-local hamiltonians.
